pkg/controllers/management/aks/aks_cluster_handler.go

```diff
@@ -107,6 +107,12 @@ func (e *aksOperatorController) onClusterChange(key string, cluster *mgmtv3.Clus
 			return cluster, err
 		}
 
+		// ensure empty availability zones are set to nil
+		cluster, requiresUpdate := pruneAvailabilityZones(cluster)
+		if requiresUpdate {
```

Comment:
Is there any reason we don't continue here? We could instead:


```
cluster, err = e.clusterClient.Update(cluster)
if err != nil {
  return cluster, err
}
```
but not sure if this choice was intentional